src/app/services/init.service.ts

```diff
@@ -0,0 +1,81 @@
+import { EnvironmentService as EnvironmentServiceAbstraction } from "jslib-common/abstractions/environment.service";
+import { EventService as EventServiceAbstraction } from "jslib-common/abstractions/event.service";
+import { I18nService as I18nServiceAbstraction } from "jslib-common/abstractions/i18n.service";
+import { NotificationsService as NotificationsServiceAbstraction } from "jslib-common/abstractions/notifications.service";
+import { PlatformUtilsService as PlatformUtilsServiceAbstraction } from "jslib-common/abstractions/platformUtils.service";
+import { StateService as StateServiceAbstraction } from "jslib-common/abstractions/state.service";
+import { SyncService as SyncServiceAbstraction } from "jslib-common/abstractions/sync.service";
+import { TwoFactorService as TwoFactorServiceAbstraction } from "jslib-common/abstractions/twoFactor.service";
+import { VaultTimeoutService as VaultTimeoutServiceAbstraction } from "jslib-common/abstractions/vaultTimeout.service";
+import { ThemeType } from "jslib-common/enums/themeType";
+import { ContainerService } from "jslib-common/services/container.service";
+import { EventService } from "jslib-common/services/event.service";
+import { VaultTimeoutService } from "jslib-common/services/vaultTimeout.service";
+
+import { I18nService } from "../../services/i18n.service";
+import { NativeMessagingService } from "../../services/nativeMessaging.service";
+
+@Injectable()
+export class InitService {
+  constructor(
+    @Inject(WINDOW) private win: Window,
+    private environmentService: EnvironmentServiceAbstraction,
+    private syncService: SyncServiceAbstraction,
+    private vaultTimeoutService: VaultTimeoutServiceAbstraction,
+    private i18nService: I18nServiceAbstraction,
+    private eventService: EventServiceAbstraction,
+    private twoFactorService: TwoFactorServiceAbstraction,
+    private notificationsService: NotificationsServiceAbstraction,
+    private platformUtilsService: PlatformUtilsServiceAbstraction,
+    private stateService: StateServiceAbstraction,
+    private cryptoService: CryptoServiceAbstraction,
+    private nativeMessagingService: NativeMessagingService
+  ) {}
+
+  init() {
+    return async () => {
+      this.nativeMessagingService.init();
+      await this.stateService.init();
+      await this.environmentService.setUrlsFromStorage();
+      this.syncService.fullSync(true);
+      (this.vaultTimeoutService as VaultTimeoutService).init(true);
+      const locale = await this.stateService.getLocale();
+      await (this.i18nService as I18nService).init(locale);
+      (this.eventService as EventService).init(true);
+      this.twoFactorService.init();
+      setTimeout(() => this.notificationsService.init(), 3000);
+      const htmlEl = this.win.document.documentElement;
+      htmlEl.classList.add("os_" + this.platformUtilsService.getDeviceString());
+      htmlEl.classList.add("locale_" + this.i18nService.translationLocale);
```

Comment:
Do you know why we are setting a class for locale?

Reply:
I don't, but here's the original git blame: https://github.com/bitwarden/desktop/commit/5fdfce0bff12a9dd594c702add7a26848d749d2b

It looks like it was added at the same time as the `os_` class, in case locale-specific styling was required. I can't see that we use it anywhere though, so I think it's safe to remove. (and I've removed it)